backend/service/github/github.go

```diff
@@ -96,9 +96,22 @@ type File struct {
 	LastModifiedSHA  string
 }
 
+type FileEntry struct {
+	Name string
+	Type string
+}
+
+type FilePath struct {
+	Path             string
+	LastModifiedTime time.Time
+	LastModifiedSHA  string
+	Files            []*FileEntry
+}
+
 // Client allows various interactions with remote repositories on GitHub.
 type Client interface {
 	GetFile(ctx context.Context, ref *RemoteRef, path string) (*File, error)
+	GetFilePath(ctx context.Context, ref *RemoteRef, path string) (*FilePath, error)
```

Comment:
What about calling this `GetDirectory`?

Reply:
+1 can we rename the structs as well?

Reply:
Renamed to GetDirectory and adjusted the structs as well.